src/core/server/saved_objects/service/lib/filter_utils.ts

```diff
@@ -205,7 +205,25 @@ export const hasFilterKeyError = (
   return null;
 };
 
-const fieldDefined = (indexMappings: IndexMapping, key: string) => {
+export const fieldDefined = (indexMappings: IndexMapping, key: string): boolean => {
   const mappingKey = 'properties.' + key.split('.').join('.properties.');
-  return get(indexMappings, mappingKey) != null;
+  const potentialKey = get(indexMappings, mappingKey);
+
+  // If the `mappingKey` does not match a valid path, before returning null,
+  // we want to check and see if the intended path was for a multi-field
+  // such as `x.attributes.field.text` where `field` is mapped to both text
+  // and keyword
```

Comment:
Really nice wording! Appreciate this fix and the late night studying it and all these tests.